com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/ApplicationColorDetector.java

```diff
@@ -100,13 +100,59 @@ private ApplicationColor getOlderApplicationColor(DeployedMta deployedMta) {
                           .orElse(null);
     }
 
-    private Phase getPhaseFromHistoricProcess(String processInstanceId) {
-        HistoricVariableInstance phaseVariableInstance = flowableFacade.getHistoricVariableInstance(processInstanceId, Variables.PHASE.getName());
-        if (phaseVariableInstance == null) {
-            return null;
-        }
+    private Optional<Operation> getLastOperationsWithFinalState(String correlationId) {
+        List<Operation> operations = getOperations(correlationId);
 
-        return Phase.valueOf((String) phaseVariableInstance.getValue());
+        Map<String, List<HistoricOperationEvent>> operationsWithHistoricEvents = operations.stream()
```

Comment:
I think this aggregation to a map first and then extracting from it is unneded.
This function can be simplified like so:
<pre>
List<> operations = getOperations();
return operations.stream().map(this::getFinalState).filter(op -> op.state != null).max(Comparator.comparing(endedAt);
</pre>
or you can even substitute the `max()` operation with `findFirst()` because `getOperations()` is already sorted
more on the `getFinalState()` changes in a later comment